homeassistant/components/camera/arlo.py

```diff
@@ -90,3 +94,30 @@ def model(self):
     def brand(self):
         """Camera brand."""
         return DEFAULT_BRAND
+
+    @property
+    def get_motion_detection_status(self):
+        """Camera Motion Detection Status."""
+        return self._motion_status
+
+    @asyncio.coroutine
+    def async_enable_motion_detect(self):
+        """Camera arm."""
+        self._motion_status = MOTION_ENABLED
+        self.hass.async_add_job(self.async_update_ha_state())
+        self.hass.async_add_job(self.async_update())
+
+    @asyncio.coroutine
+    def async_disable_motion_detect(self):
+        """Camera disarm."""
+        self._motion_status = MOTION_DISABLED
+        self.hass.async_add_job(self.async_update_ha_state())
+        self.hass.async_add_job(self.async_update())
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Perform the I/O operation with camera."""
+        if self._motion_status == MOTION_ENABLED:
+            self._base_stn.mode = ARLO_MODE_ARMED
```

Comment:
should i be creating a function to write into the base station and then yield from that function from async_update() ?